Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -319,36 +208,46 @@ private void AssignPendingVectors()
             if (innernodes.Length == 0)
                 return;
 
-            List<InventoryVector> inventoryVectors = new List<InventoryVector>();
-            uint256 result;
-            while (this.pendingInventoryVectors.TryTake(out result))
-            {
-                inventoryVectors.Add(new InventoryVector(InventoryType.MSG_BLOCK, result));
-            }
+            if (this.pendingInventoryVectors.Count == 0)
+                return;
 
+            int minHeight = int.MaxValue;
             Dictionary<int, InventoryVector> vectors = new Dictionary<int, InventoryVector>();
-            var minHeight = int.MaxValue;
-            foreach (InventoryVector vector in inventoryVectors.ToArray())
+
+            uint256 blockHash;
+            while (true)
```

Comment:
of course, i should have done it like that, i think i just did one level of refactoring and then forgot to think about next level, tx for that! it needs more memory, but it is well justified use and the number of pending vectors will never be that large for another copy to matter in memory

and i think i don't even need to dequeue, i should just .ToArray() and .Clear that